models/google_play__country_report.sql

```diff
@@ -0,0 +1,150 @@
+        
+        coalesce(install_metrics.install_events, 0) as install_events,
+        coalesce(install_metrics.uninstall_events, 0) as uninstall_events,
+        coalesce(install_metrics.update_events, 0) as update_events,    
+
+        install_metrics.total_unique_user_installs,
+        install_metrics.total_device_installs,
+        install_metrics.total_device_uninstalls,
+
+        ratings.daily_average_rating,
+        ratings.rolling_total_average_rating,
+
+        store_performance_metrics.total_store_acquisitions,
+        store_performance_metrics.total_store_visitors,
+        store_performance_metrics.store_listing_conversion_rate
+
+    from install_metrics
+    full outer join ratings
+        on install_metrics.date_day = ratings.date_day
+        and install_metrics.package_name = ratings.package_name
+        and coalesce(install_metrics.country, 'null_country') = coalesce(ratings.country, 'null_country')
+
+    full outer join store_performance_metrics
+        on store_performance_metrics.date_day = install_metrics.date_day
+        and store_performance_metrics.package_name = install_metrics.package_name
+        and coalesce(store_performance_metrics.country_region, 'null_country') = coalesce(install_metrics.country, 'null_country')
+
+), create_partitions as (
+
+    select
+        *
+
+    {%- set rolling_metrics = ['rolling_total_average_rating', 'total_unique_user_installs', 'total_device_installs', 'total_device_uninstalls', 'total_store_acquisitions', 'total_store_visitors'] -%}
+
+    {% for metric in rolling_metrics -%}
+        , sum(case when {{ metric }} is null 
+                then 0 else 1 end) over (partition by country, package_name order by date_day asc rows unbounded preceding) as {{ metric | lower }}_partition
+    {%- endfor %}
+
+    from country_join
+
+), fill_values as (
+
+    select 
+        date_day,
+        country,
+        package_name,
+        active_device_installs,
+        daily_device_installs,
+        daily_device_uninstalls,
+        daily_device_upgrades,
+        daily_user_installs,
+        daily_user_uninstalls,
+        install_events,
+        uninstall_events,
+        update_events,
+        daily_average_rating,
+        store_listing_conversion_rate
+
+        {% for metric in rolling_metrics -%}
+
+        , first_value( {{ metric }} ) over (
+            partition by {{ metric | lower }}_partition, country, package_name order by date_day asc rows between unbounded preceding and current row) as {{ metric }}
+
+        {%- endfor %}
+
+    from create_partitions
+
+), final as (
+
+    select 
+        date_day,
+        country,
+        package_name,
+        active_device_installs,
+        daily_device_installs,
+        daily_device_uninstalls,
+        daily_device_upgrades,
+        daily_user_installs,
+        daily_user_uninstalls,
+        install_events,
+        uninstall_events,
```

Comment:
Similar question as above. Not entirely sure what the difference between these fields and the daily fields?

Reply:
This may just be something we thoroughly document in our yml.